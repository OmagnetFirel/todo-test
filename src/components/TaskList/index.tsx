import React, { useState, useEffect } from "react";
import Task, { TTodo } from "../Task";
import { db } from "../../firebase";
import {
  query,
  collection,
  onSnapshot,
  doc,
  updateDoc,
  deleteDoc,
} from "firebase/firestore";

import "./index.scss";

function TaskList() {
  const [tasks, setTasks] = useState<TTodo[]>();

  useEffect(() => {
    const req = query(collection(db, "todos"));

    const unsubscribe = onSnapshot(req, (QuerySnapshot) => {
      const data = QuerySnapshot.docs.map((doc) => {
        return {
          ...doc.data(),
          id: doc.id,
        } as unknown as TTodo;
      });

      setTasks(data);
    });

    return () => {
      unsubscribe;
    };
  }, []);

  const toggleComplete = async (todo: TTodo) => {
    const todoRef = doc(db, "todos", todo.id);
    await updateDoc(todoRef, {
      completed: !todo.completed,
    });
  };

  const removeTodo = async (todo: TTodo) => {
    const todoRef = doc(db, "todos", todo.id);
    await deleteDoc(todoRef);
  };

  return (
    <ul className="taskList">
      {tasks?.map((tarefa: TTodo) => {
        return (
          <Task
            key={tarefa.id}
            todo={tarefa as TTodo}
            toggleComplete={toggleComplete}
            removeTodo={removeTodo}
          />
        );
      })}
    </ul>
  );
}

export default TaskList;
