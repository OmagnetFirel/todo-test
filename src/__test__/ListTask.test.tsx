import React from "react";
import renderer from "react-test-renderer";
import TaskList from "../components/TaskList";

describe("TaskList", () => {
  it("should render correctly", () => {
    const component = renderer.create(<TaskList />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
