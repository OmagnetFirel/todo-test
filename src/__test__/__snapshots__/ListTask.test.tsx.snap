// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList should render correctly 1`] = `
<ul>
  <li>
    sexo
  </li>
  <li>
    anal
  </li>
  <li>
    oral
  </li>
  <li>
    vaginal
  </li>
</ul>
`;
